src/blockrelay/graphene.cpp

```diff
@@ -1348,8 +1353,10 @@ void SendGrapheneBlock(CBlockRef pblock, CNode *pfrom, const CInv &inv, const CM
             uint64_t nSenderMempoolPlusBlock =
                 GetGrapheneMempoolInfo().nTx + pblock->vtx.size() - 1; // exclude coinbase
 
+            uint64_t compatibleBlockVersion = (int)std::min(
+                GRAPHENE_MAX_VERSION_SUPPORTED, pfrom->xVersion.as_u64c(XVer::BU_GRAPHENE_VERSION_SUPPORTED));
             CGrapheneBlock grapheneBlock(MakeBlockRef(*pblock), mempoolinfo.nTx, nSenderMempoolPlusBlock,
-                pfrom->xVersion.as_u64c(XVer::BU_GRAPHENE_VERSION_SUPPORTED));
+                compatibleBlockVersion, NegotiateFastFilterSupport(pfrom));
```

Comment:
But on the send side, we need to not send a graphene block (don't call this function, fall back to xthin or compact) if graphene versions are incompatible.  We don't want a throw to stop any sending at all.  But I don't see the code that skips calling SendGrapheneBlock in this PR.

Reply:
As it was, I believe an exception in SendGrapheneBlock would have resulted in a full block being sent (by the catch clause). But I added some logic to net_processing that prevents any graphene messages from being sent if sender and receiver graphene versions are incompatible. I tested it and an XThin block will be sent instead (if it's enabled).

Reply:
:+1: